solr/core/src/java/org/apache/solr/cloud/ZkConfigSetService.java

```diff
@@ -232,15 +230,20 @@ public void setConfigMetadata(String configName, Map<String, Object> data) throw
 
   @Override
   public Map<String, Object> getConfigMetadata(String configName) throws IOException {
+    byte[] data = null;
     try {
-      @SuppressWarnings("unchecked")
-      Map<String, Object> data =
-          (Map<String, Object>)
-              Utils.fromJSON(zkClient.getData(CONFIGS_ZKNODE + "/" + configName, null, null, true));
-      return data;
+      data = zkClient.getData(CONFIGS_ZKNODE + "/" + configName, null, null, true);
+    } catch (KeeperException.NoNodeException e) {
+      return new HashMap<>();
     } catch (KeeperException | InterruptedException e) {
       throw new IOException("Error getting config metadata", SolrZkClient.checkInterrupted(e));
     }
+    if (data == null) {
+      return new HashMap<>();
+    }
+    @SuppressWarnings("unchecked")
```

Comment:
likewise, should this be an if/else ?   I don't always know if stylisitically if we are okay with mulitple returns from a method?

Reply:
We have no coding standard apart from the [Google Java Style](https://google.github.io/styleguide/javaguide.html) which makes no mentions of multiple returns.  FWIW, I like multiple returns!